tephra-core/src/main/java/co/cask/tephra/distributed/ElasticPool.java

```diff
@@ -61,125 +60,100 @@
   protected abstract T create() throws E;
 
   /**
-   * A method to destroy an element. This gets called every time an element
-   * is discarded, to properly release all resources that the element might
-   * hold.
-   *
-   * @param element the element to destroy
-   */
-  protected void destroy(T element) {
-    // by default do nothing
-  }
-
-  /**
    * A method to recycle an existing element when it is returned to the pool.
    * This methods ensures that the element is in a fresh state before it can
-   * be reused by the next agent.
+   * be reused by the next agent. If the element is not to be returned to the pool,
+   * calling code is responsible for destroying it and returning false.
    *
    * @param element the element to recycle
+   * @return whether to reuse the element or not
```

Comment:
Let's make it a bit more explicit - `return true to reuse element, false to remove it from the pool`